### DISCUSSION TITLE
Bluesky Client app community fork with upstream compatibility

### DISCUSSION THREAD

**Discussion by burningtree** [bluesky-social/atproto#2998] [Bluesky Client App]
Here's an idea for a different approach to Bluesky client development: **a fork of the official Bluesky client that maintains upstream compatibility while offering enhanced features**.

The core value of this project would be its role as a community laboratory for new features and improvements. Developers could experiment with enhancements, users could test them in real-world conditions, and successful features could potentially make their way into the official client. This creates a space for innovation driven by community needs and ideas, while maintaining a stable foundation.

The project would stay in sync with the official codebase ([@bluesky-social/social-app](https://github.com/bluesky-social/social-app)), making it different from other alternative clients. This upstream compatibility ensures that users always have access to the latest core Bluesky features while enjoying community-developed enhancements.

The Bluesky ecosystem already has several alternative clients, but this approach specifically aims to bridge the gap between experimental features and official development. It's about adding value through experimentation without fragmenting the ecosystem.

This could be interesting for developers looking to contribute to a focused, practical project, and for users who want to try new features while keeping their familiar core experience. With a few dedicated developers and community support, it could be a very achievable project.

## Key principles
- Maintain strict upstream compatibility with official client while adding community features
- Focus on experimental enhancements that could benefit the whole ecosystem
- Keep same quality standards and distribute through official app stores
- Foster open development with clear path for community contributions
- Create route for successful features to be proposed upstream

## Example features
- Advanced feed customization and filtering
- Custom notification settings and filters
- Bookmarks
- Local post drafts
- Privacy-features
- App-view agnostic approach experiments
- Plugins and extensions experiments
- Custom themes and accessibility options
- Behavior changes - displaying posts from people who block you, etc.
- DID playground
- ...

----

What do you think about this idea? Does it make sense to you? Would you use this client? Would you like to contribute as a developer? Thank you for any suggestions and comments!

**Comment by DavidBuchanan314** [comment]
I think this is an "if you build it they will come" sort of thing - you could get a good start by cherrypicking from existing-but-unmerged PRs.

> Keep same quality standards

My two cents is this might actually be more of a hindrance. There's always a bit of a tradeoff between quality and rapid innovation, and I think there could be value in shifting that balance slightly (not too far, of course!) Having a slightly-hacky prototype with proven user interest is a motivator for getting a more polished implementation upstreamed.

**Comment by surfdude29** [comment]
I think this is an exciting idea and I would definitely try out this client!

**Comment by itaru2622** [comment]
This idea is nice and exciting but unfortunately it may be very tough to achieve.

especally, social-app gets several dozens new commits in every week constantly.
to care of your features, it may also need to followup other upstream components such as atproto, pds, relay(indigo).
as the results, it needs to keep following handred new commits in every week.
sometimes there may be conflicts between upstream and your new client.
while resolving conflicts, each component may get new commits. that is very tough work.

the above is my experience got weekly continuous following up almost all upstream components, for 
https://github.com/itaru2622/bluesky-selfhost-env

hope to get some breakthough, and achieve.

**Comment by mackuba** [comment]
Yeah, that would definitely be fun! I was actually thinking about doing such thing myself, making a fork with just a few patches and putting the build on my domain, and then merging upstream changes every tagged release, hopefully without too many conflicts. I was thinking I'd e.g. make it display Mastodon posts bridged via Bridgy with the full untruncated text.

I don't volunteer for a community project though, I'm too out of the loop with modern JavaScript, React, Node etc. (and don't really want to get into it) and I have way too many things on my plate already 😅
